lib-multisrc/kemono/src/eu/kanade/tachiyomi/multisrc/kemono/Kemono.kt

```diff
@@ -63,69 +64,120 @@ open class Kemono(
-                    description = PROMPT
-                    initialized = true
-                }
-            }.filterUnsupported()
-            MangasPage(creators, true).also { cacheCreators() }
-        } else {
-            fetchCreatorsPage(page) { it.apply { sortWith(comparator) } }
-        }
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> = Observable.fromCallable {
+        searchMangas(page, query, filters)
     }
 
-    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> = Observable.fromCallable {
-        if (query.isBlank()) throw Exception("Query is empty")
-        fetchCreatorsPage(page) { all ->
-            val result = all.filterTo(ArrayList()) { it.name.contains(query, ignoreCase = true) }
-            if (result.isEmpty()) return@fetchCreatorsPage emptyList()
-            if (result[0].favorited != -1) {
-                result.sortByDescending { it.favorited }
-            } else {
-                result.sortByDescending { it.updatedDate }
+    private fun searchMangas(page: Int = 1, title: String = "", filters: FilterList? = null, sortBy: Pair<String, String> = "" to ""): MangasPage {
+        var sort = sortBy
+        val typeIncluded: MutableList<String> = mutableListOf()
+        val typeExcluded: MutableList<String> = mutableListOf()
+        var fav: Boolean? = null
+        filters?.forEach { filter ->
+            when (filter) {
+                is SortFilter -> {
+                    sort = filter.getValue() to if (filter.state!!.ascending) "asc" else "desc"
+                }
+                is TypeFilter -> {
+                    filter.state.filter { state -> state.isIncluded() }.forEach { tri ->
+                        typeIncluded.add(tri.value)
+                    }
+
+                    filter.state.filter { state -> state.isExcluded() }.forEach { tri ->
+                        typeExcluded.add(tri.value)
+                    }
+                }
+                is FavouritesFilter -> {
+                    fav = when (filter.state[0].state) {
+                        0 -> null
+                        1 -> true
+                        else -> false
+                    }
+                }
+                else -> {}
             }
-            result
         }
-    }
 
-    private fun fetchCreatorsPage(
-        page: Int,
-        block: (ArrayList<KemonoCreatorDto>) -> List<KemonoCreatorDto>,
-    ): MangasPage {
-        val imgCdnUrl = this.imgCdnUrl
+        val favores = client.newCall(GET("$baseUrl/$apiPath/account/favorites", headers)).execute()
+
+        var favourites: List<KemonoFavouritesDto> = emptyList()
+        if (fav != null) {
+            if (favores.code == 401) throw Exception("You are not Logged In")
+            favourites = favores.parseAs<List<KemonoFavouritesDto>>().filterNot { it.service.lowercase() == "discord" }
+        }
+
```

Comment:
only fetch favorites if filter selected

Reply:
Hello, is that not the case in the code? Or am I interpreting your suggestion incorrectly? Thank you :pray: